src/mssw/java/com/lambdaherding/edi/mssw/ch02/ThreadLocalDateFormatter.java

```diff
@@ -0,0 +1,26 @@
+/**
+ * 
+ */
+package com.lambdaherding.edi.mssw.ch02;
+
+import java.text.SimpleDateFormat;
+import java.util.Date;
+
+import javax.swing.text.DateFormatter;
+
+/**
+ * @author Mike
+ *
+ */
+public class ThreadLocalDateFormatter {
+
+	public void printDate(){
+		ThreadLocal<DateFormatter> thread = ThreadLocal.withInitial( 
+				() -> new DateFormatter(
+						new SimpleDateFormat("dd-MMM-YYYY")
+					)	
+				);
```

Comment:
The declaration is correct. However, you miss out the power of `ThreadLocal` here by not assigning it to a static class member&mdash; if you're constructing a new one every time you hit the function, you might as well just use a `new SimpleDateFormat( ... )`.